### ISSUE TITLE
ntfy

### ISSUE THREAD

**Issue by modem7** [borgmatic-collective/docker-borgmatic#130]
This is more for discussion rather than anything else.

### **Issue 1:**

ntfy hasn't released their latest version to pypi. The latest version it installs is from 2018 (2.7.0 instead of 2.7,1).

A temporary workaround currently is to run:

Step 1:
Install latest version:
`python3 -m pip install git+https://github.com/dschep/ntfy.git@master --upgrade`

Step 2:
Install additional dependencies:
`python3 -m pip install ntfy[pid,emoji,xmpp,telegram,instapush,slack,rocketchat]`

---

### **Issue 2:**
sleekxmpp has been [sunset](https://groups.google.com/g/sleekxmpp-discussion/c/FTBMUMMvfLk), the replacement is Slixmpp. Unforfunately I don't know if this is a drop-in replacement, but I'm unsure how we'd get ntfy to utilise this. 

Note that sleekxmpp will not install on python 3.10, so we'll have issues there.

Would it be easier to drop XMPP given these issues? 

Or maybe investigate [apprise](https://github.com/caronc/apprise) instead? This would have the added benefit of replacing both ntfy and msmtp containers in favour of a single apprise container leading to less management.

**Comment by modem7** [comment]
Just had a quick look at Apprise, and it looks pretty simple to migrate to:

https://github.com/caronc/apprise#command-line

```
# Send a notification to as many servers as you want
# as you can easily chain one after another (the -vv provides some
# additional verbosity to help let you know what is going on):
apprise -vv -t 'my title' -b 'my notification body' \
   'mailto://myemail:mypass@gmail.com' \
   'pbul://o.gn5kj6nfhv736I7jC3cj3QLRiyhgl98b'

# If you don't specify a --body (-b) then stdin is used allowing
# you to use the tool as part of your every day administration:
cat /proc/cpuinfo | apprise -vv -t 'cpu info' \
   'mailto://myemail:mypass@gmail.com'

# The title field is totally optional
uptime | apprise -vv \
   'discord:///4174216298/JHMHI8qBe7bk2ZwO5U711o3dV_js'
```

Also, easy enough to install with `python3 -m pip install apprise`, so I can add any required packages to the index to speed things up too.

**Comment by modem7** [comment]
I've just added apprise and its supporting packages to the [index](https://borgmatic-collective.github.io/pypi/)

**Comment by grantbevis** [comment]
@modem7 

>Would you be able to double-check as I know you use the notification system more than I (I purely rely on healthchecks.io), and if you're happy, we can sunset both of the notification containers in favour of a new apprise one.

I don't use it anymore bud if you're happy, I'm happy

**Comment by modem7** [comment]
> @modem7
> 
> > Would you be able to double-check as I know you use the notification system more than I (I purely rely on healthchecks.io), and if you're happy, we can sunset both of the notification containers in favour of a new apprise one.
> 
> I don't use it anymore bud if you're happy, I'm happy

Yupper,  definitely happy this end - we just need to think of how to word documentation as we'd be introducing a breaking feature.

**Comment by Skydiver84de** [comment]
Hi just a quick user-feedback: just switched from ntfy to apprise with the newest base release and it´s working perfectly fine. Just getting two log lines when starting the container:

/scripts/msmtprc.sh: source: line 3: can't open '/etc/borgmatic.d/msmtp.env': No such file or directory
/scripts/env.sh: source: line 3: can't open '/etc/borgmatic.d/msmtp.env': No such file or directory

Thank you very much & pls keep up the good work!

Mario

**Comment by modem7** [comment]
> Hi just a quick user-feedback: just switched from ntfy to apprise with the newest base release and it´s working perfectly fine. Just getting two log lines when starting the container:
> 
> /scripts/msmtprc.sh: source: line 3: can't open '/etc/borgmatic.d/msmtp.env': No such file or directory
> /scripts/env.sh: source: line 3: can't open '/etc/borgmatic.d/msmtp.env': No such file or directory
> 
> Thank you very much & pls keep up the good work!
> 
> Mario
> 
> Edit:
> If I specify the tag 1.6 everything is working fine.  If using the tag :latest I end up with both ntfy and msmtp available in the container, which explains the above log lines.

Heya,

Could you post your compose snippet so we can see your config?

Cheers!

**Comment by Skydiver84de** [comment]
> > Hi just a quick user-feedback: just switched from ntfy to apprise with the newest base release and it´s working perfectly fine. Just getting two log lines when starting the container:
> > /scripts/msmtprc.sh: source: line 3: can't open '/etc/borgmatic.d/msmtp.env': No such file or directory
> > /scripts/env.sh: source: line 3: can't open '/etc/borgmatic.d/msmtp.env': No such file or directory
> > Thank you very much & pls keep up the good work!
> > Mario
> > Edit:
> > If I specify the tag 1.6 everything is working fine.  If using the tag :latest I end up with both ntfy and msmtp available in the container, which explains the above log lines.
> 
> Heya,
> 
> Could you post your compose snippet so we can see your config?
> 
> Cheers!

Hi,
I´m using it on Unraid so no docker-compose. But it was pulling the :latest tag with default settings. I think the :latest tag is pointing to the ntfy container on docker hub? As I am now pulling the :1.6 tag everything is working perfectly fine.

**Comment by modem7** [comment]
@grantbevis might be one for you

**Comment by Knapoc** [comment]
as a side note: the issue about sourcing msmtp.env in the wrong image should by solved #152.
